src/main/java/com/iridium/iridiumskyblock/listeners/BlockBreakListener.java

```diff
@@ -66,4 +68,21 @@ public void monitorBlockBreak(BlockBreakEvent event) {
         });
     }
 
+    @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
+    public void onHangingBreakByEntityEvent(HangingBreakByEntityEvent event) {
+        if (event.getRemover() instanceof Player) {
+            if (event.getEntity() instanceof ItemFrame) {
+                Player player = (Player) event.getRemover();
+                ItemFrame itemFrame = (ItemFrame) event.getEntity();
+                User user = IridiumSkyblock.getInstance().getUserManager().getUser(player);
+                Optional<Island> island = IridiumSkyblock.getInstance().getIslandManager().getIslandViaLocation(itemFrame.getLocation());
+                if (!island.isPresent()) return;
+                if (!IridiumSkyblock.getInstance().getIslandManager().getIslandPermission(island.get(), user, PermissionType.BLOCK_BREAK)) {
```

Comment:
Use ifPresent method its better

Reply:
To use it, you have to pass the plugin on Java 11 minimum, I also prefer it but cannot currently

Reply:
ah but its ifPresent

Reply:
nope its already exist on java 8 tho ?

Reply:
also we already using it on everywhere

Reply:
Ah yes, he is on Java 8 my bad,I will modify it tomorrow